site/src/component/UserReviews/UserReviews.tsx

```diff
@@ -39,14 +77,45 @@ const UserReviews: FC = () => {
         {reviews.map((review, i) => (
           <div key={`user-reviews-${i}`} className="user-reviews">
             <Divider />
-            <SubReview review={review}></SubReview>
+            <SubReview
+              review={review}
+              course={courseData.get(review.courseID)}
+              professor={professorData.get(review.professorID)}
+              editable={true}
+              editReview={editReview}
+            />
             <div className="user-reviews-footer">
-              <Button variant="danger" className="mr-3" onClick={() => deleteReview(review._id!)}>
-                Delete
-              </Button>
+              <div className="delete-review-dialog">
+                <Button variant="danger" className="mr-3" onClick={handleShow}>
+                  Delete
+                </Button>
+                <Modal show={show} onHide={handleClose} backdrop="static" keyboard={true} centered autoFocus>
+                  <Modal.Header closeButton>
+                    <Modal.Title>Delete Review Confirmation</Modal.Title>
+                  </Modal.Header>
+                  <Modal.Body>
+                    Deleting a review will remove it permanently. Are you sure you want to proceed?
+                  </Modal.Body>
+                  <Modal.Footer>
+                    <Button variant="secondary" onClick={handleClose}>
+                      Close
+                    </Button>
```

Comment:
I would say cancel instead.

```suggestion
                    <Button variant="secondary" onClick={handleClose}>
                      Cancel
                    </Button>
```